services/visual-identity/api.go

```diff
@@ -0,0 +1,75 @@
+package visualIdentity
+
+import (
+	"bufio"
+	"fmt"
+	"math/big"
+	"os"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/crypto/secp256k1"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+const (
+	// 20bytes of data described by 14 emojis requires at least 2757 length alphabet
+	alphabetLen = 2757
+	hashLen     = 14
+)
+
+func NewAPI() *API {
+	emojis, _ := loadAlphabet(alphabetLen)
+
+	return &API{
+		alphabet: &emojis,
+	}
+}
+
+type API struct {
+	alphabet *[]string
+}
+
+func (api *API) EmojiHashOf(pubkey string) (hash []string, err error) {
+	log.Info("[VisualIdentityAPI::ToEmojiHash]")
+
+	pubkeyValue, ok := new(big.Int).SetString(pubkey, 0)
+	if !ok {
+		return nil, fmt.Errorf("invalid pubkey: %s", pubkey)
+	}
+
+	x, y := secp256k1.S256().Unmarshal(pubkeyValue.Bytes())
+	if x == nil {
+		return nil, fmt.Errorf("invalid pubkey: %s", pubkey)
+	}
+
+	compressedKey := secp256k1.CompressPubkey(x, y)
+	slices, err := Slices(&compressedKey)
+	if err != nil {
+		return nil, err
+	}
+
+	emojiHashValue := new(big.Int).SetBytes(slices[1])
+
+	return ToEmojiHash(emojiHashValue, hashLen, api.alphabet)
+}
+
+func loadAlphabet(maxLen int) ([]string, error) {
+	file, err := os.Open("emojis.txt")
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	alphabet := make([]string, 0, 8)
```

Comment:
You could avoid having the slice replace the underlying array when scanner.Scan() loops, by increasing the capacity from 8 to `alphabetLen`